contracts/stellar-axelar-gateway/src/storage.rs

```diff
@@ -21,7 +14,8 @@ enum DataKey {
     #[persistent]
     #[value(MessageApprovalValue)]
     MessageApproval {
-        message_approval_key: MessageApprovalKey,
+        source_chain: String,
+        message_id: String,
```

Comment:
one thing that's a disadvantage with this approach is that the getter/setter only perform type checking but if someone passes `source_chain` and `message_id` in flipped order to them, it can lead to accidental errors. Maybe we can improve this in the future somehow, in addition to relying on unit tests